packages/client/src/pages/GamePage.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react'
+import { useState, createContext } from 'react';
 import GameLayout from '../layouts/GameLayout'
 import ActionLayer from '../layers/ActionLayer'
 import LandscapeLayer from '../layers/LandscapeLayer'
+import GamePause from '../components/gamePause'
+import GameOver from '../components/GameOver'
+import Engine from '../engine/Engine';
+
+const GameContext = createContext({});
 
 const GamePage = () => {
-  return (
-    <GameLayout>
-      <LandscapeLayer />
-      <ActionLayer />
-    </GameLayout>
+	const [pauseVisible, setPauseVisible] = useState(false);
+	const [gameOverVisible, setGameOverVisible] = useState(false);
+
+	function handlePause(){
+		setPauseVisible(true)
+	}
+	
+	function handleContinue(){
+		setPauseVisible(false)
+		const engine = Engine.get()
+		engine.pause(false)
+	}
+
+	function handleGameOver(){
+		setGameOverVisible(true)
+	}
+	
+	function handleNewGame(){
+		setGameOverVisible(false)
+		const engine = Engine.get()
+		engine.start()
+	}
```

Comment:
Так как в подавляющее большинство случаев используем стрелочные функции, для единообразия будет читабельнее и их перевести на стрелочные:
```suggestion
	const handleNewGame = () => {
		setGameOverVisible(false)

		const engine = Engine.get()
		engine.start()
	}
```

Reply:
Также, для повышения оптимизации, я бы обернул в [useCallback](https://ru.reactjs.org/docs/hooks-reference.html) (да, можно и пустое deps, и это будет полезно если дочерние компоненты часто рендерятся)
Но можно и так оставить.